src/lib/prisma-session-store.ts

```diff
@@ -1,440 +1,470 @@
+    const ms = this.options.checkPeriod;
+
+    if (typeof ms === 'number' && ms !== 0) {
+      this.stopInterval();
+      this.checkInterval = setInterval(() => {
+        this.prune();
+      }, Math.floor(ms));
+    }
+  }
+
+  /**
+   * Stop checking if sessions have expired
+   */
+  public stopInterval(): void {
+    if (this.checkInterval) clearInterval(this.checkInterval);
+  }
+
+  /**
+   * Refresh the time-to-live for the session with the given `sid`.
+   *
+   * @param sid the id of the session to refresh
+   * @param session the data of the session to resave
+   * @param callback a callback notifying that the refresh was completed
+   * or that an error occurred
+   */
+  public readonly touch = async (
+    sid: string,
+    session: PartialDeep<Express.SessionData>,
+    callback?: (err?: unknown) => void
+  ) => {
+    if (!(await this.validateConnection())) return callback?.();
+
+    const ttl = getTTL(this.options, session, sid);
+    const expires = createExpiration(ttl);
+
+    try {
+      const existingSession = await this.prisma.session.findOne({
+        where: { sid },
       });
 
-      return !this.invalidConnection;
+      if (existingSession !== null) {
+        const existingSessionData = {
+          ...this.serializer.parse(existingSession.data ?? '{}'),
+          cookie: session.cookie,
+        };
+
+        await this.prisma.session.update({
+          where: { sid: existingSession.sid },
+          data: {
+            expires,
+            data: this.serializer.stringify(existingSessionData),
+          },
+        });
+      }
+
+      // *** If there is no found session, for some reason, should it be recreated from sess *** ?
+      if (callback) defer(callback);
+    } catch (e: unknown) {
+      this.logger.error(`touch(): ${String(e)}`);
+      if (callback) defer(callback, e);
     }
   };
+
+  /**
+   * Attempts to connect to Prisma, displaying a pretty error if the connection is not possible.
+   */
+  private async connect(): Promise<void> {
+    await this.prisma?.$connect?.();
+    await this.validateConnection();
+  }
+
+  /**
+   * @description A function to generate the Prisma Record ID for a given session ID
+   *
+   * Note: If undefined and dbRecordIdIsSessionId is also undefined then a random
+   * CUID will be used instead.
+   */
+  private readonly dbRecordIdFunction = (sid: string) =>
+    this.options.dbRecordIdFunction?.(sid) ?? cuid();
+
+  /**
+   * Disables store, used when prisma cannot be connected to
+   */
+  private disable(): void {
+    this.invalidConnection = true;
+  }
+
+  /**
+   * Returns if the connect is valid or not, logging an error if it is not.
+   */
+  private async validateConnection(): Promise<boolean> {
+    await (
+      this.prisma?.$connect?.() ??
+      Promise.reject(new Error('Could not connect'))
+    ).catch(() => {
+      this.disable();
+      this.stopInterval();
+      this.logger.error(dedent`Could not connect to Sessions model in Prisma.
```

Comment:
> 'Could not connect to Sessions model in Prisma'

Wondering if this is overly precise, in a way that might be misleading... Isn't the issue that prisma can't connect _period_, rather than that prisma can't connect to the Session model _in particular_? See what you think.